pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -237,59 +263,112 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
+	}
+	url, err := url.Parse(options.MasterPublicURL)
+	if err != nil {
+		glog.Fatalf("Error parsing master public url %q: %v", options.MasterPublicURL, err)
+	}
+	genericConfig.ExternalAddress = url.Host // TODO(sttts): verify that cutting off the path is ok (it's only used as Swagger base URL)
 
-			APIResourceConfigSource: getAPIResourceConfig(options),
-			APIPrefix:               server.APIPrefix,
-			APIGroupPrefix:          server.APIGroupPrefix,
+	serviceIPRange, apiServerServiceIP, err := genericapiserver.DefaultServiceIPRange(server.GenericServerRunOptions.ServiceClusterIPRange)
+	if err != nil {
+		glog.Fatalf("Error determining service IP ranges: %v", err)
+	}
 
-			MasterCount: server.MasterCount,
+	m := &master.Config{
+		GenericConfig: genericConfig,
+		MasterCount:   server.GenericServerRunOptions.MasterCount,
 
-			// Set the TLS options for proxying to pods and services
-			// Proxying to nodes uses the kubeletClient TLS config (so can provide a different cert, and verify the node hostname)
-			ProxyTLSClientConfig: &tls.Config{
+		// Set the TLS options for proxying to pods and services
+		// Proxying to nodes uses the kubeletClient TLS config (so can provide a different cert, and verify the node hostname)
+		ProxyTransport: knet.SetTransportDefaults(&http.Transport{
+			TLSClientConfig: &tls.Config{
 				// Proxying to pods and services cannot verify hostnames, since they are contacted on randomly allocated IPs
 				InsecureSkipVerify: true,
 				Certificates:       proxyClientCerts,
 			},
+		}),
 
-			Serializer: kapi.Codecs,
+		EnableWatchCache:          server.GenericServerRunOptions.EnableWatchCache,
+		ServiceIPRange:            serviceIPRange,
+		APIServerServiceIP:        apiServerServiceIP,
+		APIServerServicePort:      443,
+		ServiceNodePortRange:      server.GenericServerRunOptions.ServiceNodePortRange,
+		KubernetesServiceNodePort: server.GenericServerRunOptions.KubernetesServiceNodePort,
 
-			EnableLogsSupport:         server.EnableLogsSupport,
-			EnableProfiling:           server.EnableProfiling,
-			EnableWatchCache:          server.EnableWatchCache,
-			MasterServiceNamespace:    server.MasterServiceNamespace,
-			ExternalHost:              server.ExternalHost,
-			MinRequestTimeout:         server.MinRequestTimeout,
-			KubernetesServiceNodePort: server.KubernetesServiceNodePort,
-		},
+		StorageFactory: storageFactory,
 
 		EventTTL: server.EventTTL,
 
-		KubeletClient: kubeletClient,
+		KubeletClientConfig: *configapi.GetKubeletClientConfig(options),
 
+		EnableLogsSupport:     false, // don't expose server logs
 		EnableCoreControllers: true,
 
-		DeleteCollectionWorkers: server.DeleteCollectionWorkers,
+		DeleteCollectionWorkers: server.GenericServerRunOptions.DeleteCollectionWorkers,
 	}
 
-	if server.EnableWatchCache {
-		cachesize.SetWatchCacheSizes(server.WatchCacheSizes)
+	if m.EnableWatchCache {
+		cachesize.SetWatchCacheSizes(server.GenericServerRunOptions.WatchCacheSizes)
```

Comment:
We don't do `cachesize.InitializeWatchCacheSizes(s.GenericServerRunOptions.TargetRAMMB)` as upstream. Is this intentional?